packages/bpk-component-carousel/src/BpkCarousel.js

```diff
@@ -0,0 +1,169 @@
+import ArrowRightIcon from '../../bpk-component-icon/sm/long-arrow-right';
+
+import STYLES from './BpkCarousel.scss';
+
+const AlignedArrowLeft = withButtonAlignment(withRtlSupport(ArrowLeftIcon));
+const AlignedArrowRight = withButtonAlignment(withRtlSupport(ArrowRightIcon));
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  className: ?string,
+  wrapperClassName: ?string,
+  itemClassName: ?string,
+  children: ?any,
+};
+
+type State = {
+  position: number,
+  nextArrowDisabled: boolean,
+  prevArrowDisabled: boolean,
+};
+
+class BpkCarousel extends Component<Props, State> {
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node,
+  };
+
+  static defaultProps = {
+    className: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      position: 0,
+      nextArrowDisabled: false,
+      prevArrowDisabled: true,
+    };
+  }
+
+  getOrder(itemIndex) {
+    const { position } = this.state;
+    const { children } = this.props;
+    const numItems = children.length;
+    if (itemIndex - position < 0) {
+      return numItems - Math.abs(itemIndex - position);
+    }
+    return itemIndex - position;
+  }
+
+  nextSlide = () => {
+    const { position } = this.state;
+    const { children } = this.props;
+    const numItems = children.length || 1;
+    this.doSliding(position === numItems - 1 ? 0 : position + 1);
+  };
+
+  prevSlide = () => {
+    const { position } = this.state;
+    const { children } = this.props;
+    const numItems = children.length || 1;
+    this.doSliding(position === 0 ? numItems - 1 : position - 1);
+  };
+
+  doSliding = position => {
+    this.setState({
+      position,
+      nextArrowDisabled: position === this.props.children.length - 1,
+      prevArrowDisabled: position === 0,
+    });
+  };
+
+  render() {
+    const {
+      className,
+      wrapperClassName,
+      itemClassName,
+      children,
+      ...rest
+    } = this.props;
+    const { prevArrowDisabled, nextArrowDisabled } = this.state;
+    const classNames = getClassName('bpk-carousel', className);
+    const wrapperClassNames = getClassName(
+      'bpk-carousel__wrapper',
+      wrapperClassName,
+    );
+    const itemClassNames = getClassName('bpk-carousel__item', itemClassName);
+    const normalisedChildren = Array.isArray(children) ? children : [children];
+
+    return (
+      <div className={classNames} {...rest}>
+        <div>
+          <BpkButton
+            iconOnly
+            disabled={prevArrowDisabled}
+            onClick={() => this.prevSlide()}
+          >
+            <AlignedArrowLeft />
+          </BpkButton>
+        </div>
+        <div className={wrapperClassNames}>
+          <div className={getClassName('bpk-carousel__container')}>
+            {normalisedChildren.map((child, index) => (
+              <div
+                className={itemClassNames}
+                id={index}
```

Comment:
If you use `React.Children.toArray(children)` this is most likely not necessary

Reply:
Which part is not necessary?

Reply:
Sorry I read `key` instead of `id`. Key would not be necessary :)